iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -193,99 +196,159 @@ TEST_F(UnixDomainSocket_test, SendOnServerLeadsToError)
     sendOnServerLeadsToError([&](auto& msg) { return server.send(msg); });
 }
 
-void successfulSendAndReceive(const std::string& message, const sendCall_t& send, const receiveCall_t& receive)
+void successfulSendAndReceive(const std::vector<std::string>& messages,
+                              const sendCall_t& send,
+                              const receiveCall_t& receive)
 {
-    ASSERT_FALSE(send(message).has_error());
+    for (auto& m : messages)
+    {
+        ASSERT_FALSE(send(m).has_error());
+    }
 
-    auto receivedMessage = receive();
-    ASSERT_FALSE(receivedMessage.has_error());
-    EXPECT_EQ(message, *receivedMessage);
+    std::vector<std::string> receivedMessages;
+    for (uint64_t i = 0, messagesSize = messages.size(); i < messagesSize; ++i)
+    {
+        auto receivedMessage = receive();
+        ASSERT_FALSE(receivedMessage.has_error());
+        receivedMessages.emplace_back(*receivedMessage);
+    }
+    EXPECT_EQ(messages, receivedMessages);
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfNonEmptyMessageWithSendAndReceive)
 {
     successfulSendAndReceive(
-        "what's hypnotoads eye color?",
+        {"what's hypnotoads eye color?"},
         [&](auto& msg) { return client.send(msg); },
         [&]() { return server.receive(); });
 }
 
 TEST_F(UnixDomainSocket_test, SuccessfulCommunicationOfNonEmptyMessageWithTimedSendAndReceive)
 {
     successfulSendAndReceive(
-        "the earth is a disc on the back of hypnotoad",
+        {"the earth is a disc on the back of hypnotoad"},
```

Comment:
This is not true! It's a disk on the back of elephants on the back of hypnotoad. Didn't you study the gospels ;)